scrnaseq.Rmd

```diff
@@ -159,189 +331,13 @@ invisible(knitcitations::citep(citation("knitr")))
-# condition_group2: Condition levels in group 2, multiple levels concatenated by the plus character
-#                     Empty string = all levels not in group1 (cannot be used if group1 is empty)
-#
-# Optional columns:
-# subset_column: Categorial column in the cell metadata to subset before testing (default: NA)
-#                  Specify "orig.ident" for sample and "seurat_clusters" for cluster 
-# subset_group: Further subset levels (default: NA)
-#                 For the individual analysis of multiple levels separate by semicolons
-#                 For the joint analysis of multiple levels concatenate by the plus character 
-#                 For the individual analysis of all levels empty string ""
-# assay: Seurat assay to test on; can also be a Seurat dimensionality reduction (default: "RNA")
-# slot: In case assay is a Seurat assay object, which slot to use (default: "data")
-# padj: Maximum adjusted p-value (default: 0.05)
-# log2FC: Minimum absolute log2 fold change (default: 0)
-# min_pct: Minimum percentage of cells expressing a gene to test (default: 0.1)
-# test: Type of test; "wilcox", "bimod", "roc", "t", "negbinom", "poisson", "LR", "MAST", "DESeq2"; (default: "wilcox")
-# downsample_cells_n: Downsample each group to at most n cells to speed up tests (default: NA)
-# latent_vars: Additional variables to account for; multiple variables need to be concatenated by semicolons; will overwrite the default by param$latent_vars (default: none).
-param$deg_contrasts = data.frame(condition_column=c("orig.ident", "orig.ident", "Phase"),
-                                 condition_group1=c("pbmc_10x", "pbmc_10x", "G1"),
-                                 condition_group2=c("pbmc_smartseq2_sample1", "pbmc_smartseq2_sample1", "G2M"),
-                                 subset_column=c(NA, "seurat_clusters", "seurat_clusters"),
-                                 subset_group=c(NA, "", "1;2"),
-                                 downsample_cells_n=c(NA, 50, 30),
-                                 log2FC=log2(c(1.5, 1.5, 1.5)))
-
-# P-value threshold for functional enrichment tests
-param$enrichr_padj = 0.05
-
-# Enrichr databases of interest
-param$enrichr_dbs = c("GO_Molecular_Function_2018", "GO_Biological_Process_2018", "GO_Cellular_Component_2018")
-
-######################
-# General parameters #
-######################
-# Main colour to use for plots
-param$col = "palevioletred"
-
-# Colour palette used for samples
-param$col_palette_samples = "ggsci::pal_jama"
-
-# Colour palette used for cluster
-param$col_palette_clusters = "ggsci::pal_igv"
-
-# Path to git repository
-param$path_to_git = "."
-
-# Debugging mode: 
-# 'default_debugging' for default, 'terminal_debugger' for debugging without X11, 'print_traceback' for non-interactive sessions 
-param$debugging_mode = "default_debugging"
-```
-
 ```{r preparation_and_initial_checks}
+# Copy input parameters to internal parameter list
```

Comment:
I'd have to have a closer look but doesn't `functions_util.R` take care of this?

For example: 
```
  # Check samples_to_drop
  if ("samples_to_drop" %in% names(param) && !is.character(param$samples_to_drop)) {
    error_messages = c(error_messages, "The parameter 'samples_to_drop' is not a character vector. Please set to NULL if there are no samples to drop!")
  }
```